libraries/AP_Arming/AP_Arming.cpp

```diff
@@ -855,6 +856,15 @@ bool AP_Arming::mission_checks(bool report)
         check_failed(ARMING_CHECK_MISSION, report, "Failed to open %s", AP_MISSION_SDCARD_FILENAME);
     }
 #endif
+
+    // do not allow arming if there are no mission items and we are in
+    // (e.g.) AUTO mode
+    if (AP::vehicle()->current_mode_requires_mission() &&
+        (mission == nullptr || mission->num_commands() <= 1)) {
+        check_failed(ARMING_CHECK_MISSION, report, "Mode requires non-empty mission");
```

Comment:
```suggestion
        check_failed(ARMING_CHECK_MISSION, report, "Mode requires mission");
```
I think "non-empty" is redundant.